### DISCUSSION TITLE
connection refused on port 4000

### DISCUSSION THREAD

**Discussion by raducadar** [teslamate-org/teslamate#3215] [Q&A]
The installations steps are quite simple, no issue during install, but something is not working as expected and impossible to find the issue.

root@NAS:/volume5/@docker# docker ps
CONTAINER ID   IMAGE                        COMMAND                  CREATED         STATUS          PORTS                    NAMES
310801175a40   teslamate/teslamate:latest   "tini -- /bin/sh /en…"   4 minutes ago   Up 4 minutes    0.0.0.0:4000->4000/tcp   docker_teslamate_1
2413b19688cc   eclipse-mosquitto:2          "/docker-entrypoint.…"   2 hours ago     Up 41 minutes   1883/tcp                 docker_mosquitto_1
78111188e225   teslamate/grafana:latest     "/run.sh"                2 hours ago     Up 41 minutes   0.0.0.0:3000->3000/tcp   docker_grafana_1
7adcc82fea9c   postgres:14                  "docker-entrypoint.s…"   2 hours ago     Up 41 minutes   5432/tcp                 docker_database_1
root@NAS:/volume5/@docker# curl -v 192.168.178.44:4000
* Failed to connect to 192.168.178.44 port 4000: Connection refused
curl: (7) Failed to connect to 192.168.178.44 port 4000: Connection refused
root@NAS:/volume5/@docker# curl -v 192.168.178.44:3000
> GET / HTTP/1.1
> Host: 192.168.178.44:3000
> User-Agent: curl/7.54.0
> Accept: */*
>
< HTTP/1.1 302 Found
< Cache-Control: no-cache
< Content-Type: text/html; charset=utf-8
< Expires: -1
< Location: /login
< Pragma: no-cache
< Set-Cookie: redirect_to=%2F; Path=/; HttpOnly; SameSite=Lax
< X-Content-Type-Options: nosniff
< X-Xss-Protection: 1; mode=block
< Date: Wed, 24 May 2023 18:53:16 GMT
< Content-Length: 29
<
<a href="/login">Found</a>.

root@NAS:/volume5/@docker#

somehow the connection is refused on port 4000

any ideas?

**Comment by katermikesch777** [comment]
Tunnel remote localhost:4000 through SSH to your local machine:
ssh -L 4000:192.168.178.44:4000 root@192.168.178.44:21
or use something like MobaXTerm/Putty on Windows to setup the Tunnel

Then you can access with your local browser, just type localhost:4000 into it and it will get forwarded to remote host to port 4000 (teslamate login)

Can you access port 3000 (Grafana) with 192.168.178.44:3000?

I have the same "issue" and solved it with tunneling the port, but i dont know if its intended that you can access the TeslaMate configuration only on localhost. Grafana is working with ip, atleast for me. After initial login and setup i could access without forwarding port to local machine
